apis/install/v1alpha1/common.go

```diff
@@ -93,4 +100,37 @@ type CommonSpecBase struct {
 	AdditionalVolumes []corev1.Volume `json:"additionalVolumes,omitempty"`
 	// Additional volume mounts that are added as volumes
 	AdditionalVolumeMounts []corev1.VolumeMount `json:"additionalVolumeMounts,omitempty"`
+	// PortConfig is automatically populated with defaults and overlaid by values in ApplicationConfig.
+	PortConfig PortConfig `json:"portConfig,omitempty"`
+}
+
+// BuildPortConfig extracts ports from the ApplicationConfig and applies inplace
+// as the PortConfig
+func (c *CommonSpecBase) BuildPortConfig() error {
+	appConfig, err := convertRawExtensionToYaml(c.ApplicationConfig)
+	if err != nil {
+		return err
+	}
+	// defaults
```

Comment:
We probably shouldn't have defaults here.

Reply:
why not?

Reply:
Also, the controller for a given service creates the port configuration -- this is just a container for storing it. so having an httpPort default will not expose that automatically unless the controller is explicitly creating an http Deployment, Service, etc.